gigareview/common/tests/unit/ManuscriptTest.php

```diff
@@ -32,4 +32,66 @@ public function testCanCreateTimestamp()
+            [
+                'manuscript_number' => 'GIGA-D-22-00054',
+                'article_title' => 'A machine learning framework for discovery and enrichment of metagenomics metadata from open access publications',
+                'editorial_status_date' => '6/7/2022',
+                'editorial_status' => 'Final Decision Accept'
+            ]
+        ];
+
+        $manuscriptInstance = Manuscript::createInstancesFromEmReport($sampleCsvReportData);
+        $this->assertNotNull($manuscriptInstance);
+        $this->assertIsArray($manuscriptInstance);
+
+        foreach ($manuscriptInstance as $instance) {
+            $this->assertInstanceOf('common\models\Manuscript', $instance, "Manuscript Instance is not instantiated!");
+            $this->assertArrayHasKey('manuscript_number', $instance, "Key is not found in Manuscript instance!");
+            $this->assertArrayHasKey('article_title', $instance, "Key is not found in Manuscript instance!");
+            $this->assertArrayHasKey('editorial_status_date', $instance, "Key is not found in Manuscript instance!");
+            $this->assertArrayHasKey('editorial_status', $instance, "Key is not found in Manuscript instance!");
+            $this->assertEquals($sampleCsvReportData[0]['manuscript_number'], $instance->manuscript_number, "Value is not matched in Manuscript instance!");
+            $this->assertEquals($sampleCsvReportData[0]['article_title'], $instance->article_title, "Value is not matched in Manuscript instance!");
+            $this->assertEquals($sampleCsvReportData[0]['editorial_status_date'], $instance->editorial_status_date, "Value is not matched in Manuscript instance!");
+            $this->assertEquals($sampleCsvReportData[0]['editorial_status'], $instance->editorial_status, "Value is not matched in Manuscript instance!");
+        }
+    }
+
+    public function testCanCreateInstancesFromReportWithTwoEntries()
+    {
+        $sampleCsvReportData = [
+            [
+                'manuscript_number' => 'GIGA-D-22-00054',
+                'article_title' => 'A machine learning framework for discovery and enrichment of metagenomics metadata from open access publications',
+                'editorial_status_date' => '6/7/2022',
+                'editorial_status' => 'Final Decision Accept'
+            ],
+            [
+                'manuscript_number' => 'GIGA-D-22-00060',
+                'article_title' => 'A chromosome-level genome of the booklouse, Liposcelis brunnea provides insight into louse evolution and environmental stress adaptation',
+                'editorial_status_date' => '6/7/2022',
+                'editorial_status' => 'Final Decision Reject'
+            ],
+        ];
+
+        $manuscriptInstances = Manuscript::createInstancesFromEmReport($sampleCsvReportData);
+        $this->assertNotNull($manuscriptInstances);
+        $this->assertIsArray($manuscriptInstances);
+
+        for ($i=0; $i <= count($sampleCsvReportData) - 1; $i++) {
```

Comment:
Could use [`array_map`](https://riptutorial.com/php/example/6400/iterating-multiple-arrays-together) to iterate 2 arrays together.

Reply:
Just fyi:

For the test `ManuscriptTest.php:testCanCreateInstancesFromReportWithTwoEntries`,

`for loop` takes `Time: 362 ms, Memory: 12.00 MB` to finish, while `array_map()` takes `Time: 363 ms, Memory: 12.00 MB`, it seems there is significant improvement.